documentation/components/markdown/editor-component/Cards.tsx

```diff
@@ -0,0 +1,154 @@
+import {
+  Image,
+  Text,
+  Grid,
+  Box,
+  Heading,
+  Stack,
+  ChipToggleGroup,
+  SearchInput,
+  styled
+} from '@atom-learning/components'
+import { intersection } from "lodash";
+import * as React from 'react'
+import { debounce } from 'throttle-debounce'
+
+type TCardsItemProps = { image: string, heading: string, description: string, link: { href: string, isExternal: boolean }, tags: string | string[] }
+
+const CardsItem: React.FC<TCardsItemProps> = ({ image, heading, description, link }) => {
+  const isLink = !!link?.href;
+  const elementSpecificProps = isLink
+    ? {
+      as: 'a',
+      href: link.href,
+      target: link.isExternal ? '_blank' : '_self',
+      rel: link.isExternal ? 'noopener noreferrer nofollow' : false
+    }
+    : { as: 'div' }
+
+  return (
+    <Box {...elementSpecificProps} css={{ color: 'inherit', textDecoration: 'none', display: 'block' }}>
+      <Box css={{
+        background: '$base1',
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+        mb: '$2',
+        p: '$2',
+        aspectRatio: "16/9"
+      }}>
+        {image && <Image src={image} alt='' css={{ maxHeight: '100%' }} />}
+      </Box>
+      <Heading as="h3" size="xs" css={{ mb: '$3' }}>{heading}</Heading>
+      <Text size="xs">{description}</Text>
+    </Box >
+  )
+}
+
+const StyledUl = styled('ul', {
+  p: 0,
+  m: 0,
+})
+
+type CardsProps = {
+  showSearch: boolean
+  showTagsFilter: boolean
+  items: TCardsItemProps[]
+}
+
+const checkIfMatchingTags = (itemTags, selectedTags) => {
+  const isAllValueSelected = selectedTags?.includes('')
+  if (isAllValueSelected) return true;
+  if (typeof itemTags === 'string') { return selectedTags?.includes(itemTags) }
+  if (Array.isArray(itemTags)) { return intersection(itemTags, selectedTags).length }
+  return false;
+}
+
+const checkIfMatchingSearch = (item, searchValue) => {
+  const itemStringified = JSON.stringify(item)
+  return itemStringified.toLowerCase().includes(searchValue.toLowerCase())
+}
+
+export const Cards: React.FC<CardsProps> = ({ showSearch, showTagsFilter, items }) => {
+  const [searchValue, setSearchValue] = React.useState('')
+  const handleSetSearchValue = React.useCallback(debounce(500, (value) => setSearchValue(value.toLowerCase())), [])
+
+
+  const [selectedTags, setSelectedTags] = React.useState([''])
+  const handleSelectedTagsValueChange = React.useCallback((newSelectedTags) => {
+    // "All" unselects all other tags
+    const hasJustToggledAll = newSelectedTags[newSelectedTags.length - 1] === ''
+    if (hasJustToggledAll) return setSelectedTags([''])
+
+    // Always keep "All" selected as a fallback if no value selected
+    if (newSelectedTags.length === 0) return setSelectedTags([''])
+
+    // Remove "All" from the tags array if other values have been selected
+    if (newSelectedTags.length > 1) return setSelectedTags(newSelectedTags.filter(Boolean))
+    setSelectedTags(newSelectedTags)
```

Comment:
I think a space here will make it clearer that the final `setSelectedTags` call isn't related to this conditional
```suggestion
    if (newSelectedTags.length > 1) return setSelectedTags(newSelectedTags.filter(Boolean))
    
    setSelectedTags(newSelectedTags)
```